core/domain/stats_jobs_one_off.py

```diff
@@ -195,3 +196,248 @@ def reduce(key, stringified_values):
                     'all: %s sum:%s' % (
                         key, state_name, all_state_hit[state_name],
                         sum_state_hit[state_name]),)
+
+
+class AnswerModelInteractionConsistencyJob(jobs.BaseMapReduceJobManager):
+    """A one-off job to check the consistency of internal answer model state.
+
+    Verifies that all (exploration_id, exploration_version, state_name) tuples
+    correspond to exactly one interaction ID across all submitted answer
+    buckets. This job is used as a preliminary audit before running
+    PurgeNewAnswersJob to ensure that job does not leave any remaining buckets
+    after purging.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [stats_models.StateAnswersModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function. Must be declared @staticmethod.
+
+        Args:
+            item: StateAnswersModel.
+
+        Yields:
+            tuple. a 2-tuple in the form
+                ('exploration_id-exploration_version-state_name', value).
+                'exploration_id': str. the id of the exploration.
+                'exploration_version': number. the version of the exploration.
+                'state_name': str. the name of the corresponding state.
+                'value': a dict, whose structure is as follows:
+                    {
+                        'exploration_id': str. the id of the exploration.
+                        'exploration_version': number. the version of the
+                            exploration.
+                        'state_name': str. the name of the corresponding state.
+                        'interaction_id': str. the corresponding interaction ID.
+                    }
+        """
+        combined_key = '%s-%s-%s' % (
```

Comment:
Oh, but you never decode it. I guess theoretically you could still get a clash but it might be OK?